api/client/client.go

```diff
@@ -116,15 +116,20 @@ func newClient(cfg Config, dialer ContextDialer, tlsConfig *tls.Config) *Client
 // no dialer is given, the first address will be used. This address must
 // be an auth server address.
 func connectInBackground(ctx context.Context, cfg Config) (*Client, error) {
-	tlsConfig, err := cfg.Credentials[0].TLSConfig()
+	var err error
+	params := connectParams{cfg: cfg}
+
+	params.tlsConfig, err = cfg.Credentials[0].TLSConfig()
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	if cfg.Dialer != nil {
-		return dialerConnect(ctx, cfg, cfg.Dialer, tlsConfig)
+		params.dialer = cfg.Dialer
+		return dialerConnect(ctx, params)
 	} else if len(cfg.Addrs) != 0 {
-		return authConnect(ctx, cfg, cfg.Addrs[0], tlsConfig)
+		params.addr = cfg.Addrs[0]
```

Comment:
This is more of a stylistic nit, but I've found that when you're first creating a struct ("params" in this case) and then keep filling it with fields, it's a bit more confusing for the reader because you cannot easily see all fields that are eventually being passed. So if I was writing this, i would probably do it like this:

```go
tlsConfig, err := cfg.Credentials[0].TLSConfig()
if err != nil {
  return nil, trace.Wrap(err)
}
if cfg.Dialer != nil {
  return dialerConnect(ctx, connectParams{
    cfg: cfg,
    tlsConfig: tlsConfig,
    dialer: cfg.Dialer,
  })
} else if len(cfg.Addrs) != 0 {
  return authConnect(ctx, connectParams{
    cfg: cfg,
    tlsConfig: tlsConfig,
    addr: cfg.Addrs[0],
  })
}
```